test/api/encoding.js

```diff
@@ -11,8 +12,14 @@ function fixturePath(fixture) {
   return path.resolve(__dirname, "fixtures/encoding", fixture);
 }
 
+// This can't use `fs.promises`, as it has a bug with non-UTF-8 encodings:
+let readFile;
 function readFixture(fixture) {
-  return fs.readFile(fixturePath(fixture));
+  if (!readFile) {
```

Comment:
The extra code complexity here of caching detracts from readability too much. Just do `return util.promisify(fs.readFile)(...)`.